google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -1357,4 +1359,22 @@ private static GrpcStatusCode statusCodeFor(StatusCode.Code code) {
         throw new IllegalStateException("Unrecognized status code: " + code);
     }
   }
+
+  private static ImmutableList<String> updateFields() {
```

Comment:
This isn't ideal but we need a way to create the field mask with only the appropriate fields so that we are not attempting to update immutable fields causing an exception. For Json since it uses patch we do not have to worry about this but GRPC does not appear to have an equivalent.

Reply:
We tried pairing to see if there's an alternative but the available options we tried were not successful. 

Specifically we tried:

```
.filter(f -> f.getKey().toProto().getOptions().getAllFields().entrySet.contains(FieldBehavior.IMMUTABLE))
```

```
.filter(f -> f.getKey().toProto().getOptions().getAllFields().containsValue(FieldBehavior.IMMUTABLE))
```

We are probably not using the correct type to determine equality; but we shall look at this another time.

@sydney-munro could you add a todo for now?

Reply:
I put the todo in the updateMaskGenerator if thats alright.